account/src/use-case/createUserAccount.js

```diff
@@ -1,16 +1,25 @@
-import  bcrypt from 'bcryptjs';
-import { saveAccount } from "../repositories/accountRepository.js";
+import bcrypt from "bcryptjs";
+import {
+  existsUserByEmail,
+  saveAccount,
+} from "../repositories/accountRepository.js";
 
 export async function createUserUseCase(name, email, password) {
-    const createdDate = new Date().toISOString().substring(0, 10);
-    const passwordHash = bcrypt.hashSync(password ,10);
-    const user = {
-        name,
-        email,
-        password: passwordHash,
-        createdDate
+  const userAlreadyExists = await existsUserByEmail(email);
 
-    };
-    await saveAccount(user);
-    return user;
- }
\ No newline at end of file
+  if (userAlreadyExists) {
+    console.error("e-mail is already registered in our base.", email);
+    throw new Error("e-mail is already registered in our base.");
```

Comment:
pode remover o `in our base`. Dizer que o usuário já tá cadastrado é claro o suficiente.

Reply:
Feito.